flutter/lib/resources/resource_manager.dart

```diff
@@ -167,13 +198,33 @@ class ResourceManager {
     _batchPresets = result;
   }
 
-  Future<List<String>> validateResourcesExist(List<String> resources) async {
+  Future<List<String>> validateResourcesExist(List<Resource> resources) async {
     final missingResources = <String>[];
     for (var r in resources) {
-      if (!await isResourceExist(r)) {
-        missingResources.add(r);
+      if (!await isResourceExist(r.path)) {
+        missingResources.add(r.path);
       }
     }
     return missingResources;
   }
+
+  Future<List<Resource>> validateResourcesChecksum(
+      List<Resource> resources) async {
+    final checksumFailedResources = <Resource>[];
+    for (final resource in resources) {
+      final md5Checksum = resource.md5Checksum;
+      if (md5Checksum == null || md5Checksum.isEmpty) continue;
+      String? localPath;
+      if (cacheManager.isResourceAnArchive(resource.path)) {
+        localPath = cacheManager.getArchive(resource.path);
+      } else {
+        localPath = cacheManager.get(resource.path);
+      }
+      if (localPath == null || !File(localPath).existsSync()) continue;
```

Comment:
Let's use the async `exists` if possible.

Reply:
ok